Packs/ParseHTMLTables/Scripts/ParseHTMLTables/ParseHTMLTables.py

```diff
@@ -15,29 +14,119 @@ def __init__(self, title: str):
+        Add a row with cells and labels.
+
+        :param columns: List of data cells of the row.
+        :param labels: List of header cells of the row.
+        """
+        rowspan_labels = self.__rowspan_labels
+
+        # Normalize labels
+        if labels and any(label.attrs.get('rowspan') for label in labels):
+            self.__set_rowspan_labels(labels)
+
+        normalized_labels = []
+        if labels:
+            for i, (count, label) in enumerate(rowspan_labels):
+                if count >= 2:
+                    normalized_labels.append(label)
+
+            for label in labels:
+                try:
+                    colspan = int(label.attrs.get('colspan') or 1)
+                except Exception:
+                    colspan = 1
+                normalized_labels += [label.text.strip()] * max(1, colspan)
+
+        # Normalize columns
+        if any(col.attrs.get('rowspan') for col in columns):
+            self.__set_rowspan_labels(columns)
+
+        normalized_columns = []
+        for i, (count, label) in enumerate(rowspan_labels):
+            if count >= 2:
+                normalized_columns.append(label)
+                rowspan_labels[i] = count - 1, label
+
+        for col in columns:
+            try:
+                colspan = int(col.attrs.get('colspan') or 1)
+            except Exception:
+                colspan = 1
+            normalized_columns += [col.text.strip()] * max(1, colspan)
+
+        self.__rows.append((normalized_labels, normalized_columns))
 
     def get_rows(self) -> List[Tuple[List[str], List[str]]]:
         return self.__rows
 
-    def make_pretty_table_rows(self) -> Any:
+    def make_pretty_table_rows(self, default_header_line: Optional[str] = None) -> Any:
+        """
+        Format a table
+
+        :param default_header_line: Which table line handles as header by default, 'first_column' or 'first_row'
+        :return: The table formatted in JSON structure.
+        """
         rows: List[Union[str, Dict[str, Any]]] = []
         temp_row: Dict[str, Any] = {}
 
-        for labels, cols in self.__rows:
+        tbl_rows = self.__rows
+        headers = self.__headers
+
+        if default_header_line and default_header_line != 'none':
+            if not headers and not any(labels for labels, cols in tbl_rows):
+                if default_header_line == 'first_column':
+                    # The first column is considered as header
+                    tbl_rows = [([], list(cols)) for cols in zip(*[cols for labels, cols in tbl_rows])]  # transpose
+                    if tbl_rows and tbl_rows[0][1]:
+                        headers = tbl_rows[0][1]
+                        tbl_rows = tbl_rows[1:]
+                elif default_header_line == 'first_row':
+                    # The first row is considered as header
+                    if tbl_rows and tbl_rows[0][1]:
+                        headers = tbl_rows[0][1]
+                        tbl_rows = tbl_rows[1:]
```

Comment:
Suppose to be identical to the if statement in line 116? if so, why are they duplicated?

Reply:
OK, I've removed the redundancy for more sophisticated code.